demo_nodes_cpp/src/events/matched_event_detect.cpp

```diff
@@ -0,0 +1,223 @@
+// Copyright 2023 Sony Group Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <string>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "std_msgs/msg/string.hpp"
+
+#include "demo_nodes_cpp/visibility_control.h"
+
+using namespace std::chrono_literals;
+
+// This demo program shows matched event work.
+// Matched event occurs when publisher and subscription establishes the connection.
+// Class MatchedEventDetectNode output connection information of publisher and subscription.
+// Class MultiSubNode is used to create/destroy subscription to connect/disconnect the publisher of
+// MatchedEventDetectNode.
+// Class MultiPubNode is used to created/destroy publisher to connect/disconnect the subscription
+// of MatchedEventDetectNode.
+
+class MatchedEventDetectNode : public rclcpp::Node
+{
+public:
+  DEMO_NODES_CPP_PUBLIC explicit MatchedEventDetectNode(
```

Comment:
If using the `main` in this file, which means this file will not be built into a library, I think it's unnecessary to add `DEMO_NODES_CPP_PUBLIC`.

Reply:
Yes.